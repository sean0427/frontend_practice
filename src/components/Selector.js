import React from 'react';

export default class Selector extends React.Component {
    static defaultProps = {
        textName: 'text',
        valueName: 'vaule',
        name: 'title',
        options: [],
    };

    constructor(props) {
        super(props);

        this.state = {
            selected: '',
        };

        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.setState({ selected: event.target.value });
        event.preventDefault();
    }

    render() {
        const { options, valueName, textName, name } = this.props;

        return (
            <label>
                <select
                    name={ name }
                    value={ this.state.selected }
                    onChange={ this.handleChange }
                >
                    {
                        options.map((value, index) =>
                            <option
                                key={ index }
                                value={ value[valueName] }
                            >
                                { value[textName] }
                            </option>
                        )

                    }
                </select>
            </label>
        );
    }
}
