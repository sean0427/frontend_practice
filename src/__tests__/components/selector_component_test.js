import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Selector from '../../components/Selector';

const TEXT = 'TEXT';
const VALUE = 'VALUE';
const ITEM_COUNT = 2;

const TEST_PROPS = {
    name: TEXT,
    textName: TEXT,
    valueName: VALUE,
    options: [
        { TEXT: 1, VALUE: 'a' },
        { TEXT: 2, VALUE: 'b' },
    ],
};

describe('Test Component Product', () => {
    it('render defualt', () => {
        const component = shallow(<Selector />);

        expect(component.find('label')).toHaveLength(1);
        expect(component.find('select')).toHaveLength(1);
        expect(component.find('option')).toHaveLength(0);
    });

    it('render with props input', () => {
        const component = shallow(<Selector />);

        component.setProps(TEST_PROPS);
        const options = component.find('option');
        expect(options).toHaveLength(ITEM_COUNT);
    });

    it('select chnage', () => {
        const component = mount(<Selector {...TEST_PROPS } />);
        component.find('select').simulate('change', { target: { value: 'b' }});
        expect(component.find('select').props().value).toEqual('b');
    });

    it('check change by snapshot', () => {
        const json = renderer.create(<Selector />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
