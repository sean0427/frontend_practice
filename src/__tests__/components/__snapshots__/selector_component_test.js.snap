// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Product check change by snapshot 1`] = `
<label>
  <select
    onChange={[Function]}
    value=""
  />
</label>
`;
